services/businessconfig/src/main/java/org/opfab/businessconfig/services/ProcessesService.java

```diff
@@ -659,10 +663,14 @@ public synchronized void updateBusinessDataFile(String fileContent, String resou
                 Files.createDirectories(businessDataPath);
             } catch (IOException e) {
                 log.error("Impossible to create the necessary folder", businessDataPath.toString(), e);
+                throw e;
             }
         }
 
-        this.isResourceJSON(fileContent);
+        if (!this.isResourceJSON(fileContent)) {
+            log.error("Invalid JSON format in the file content for resource: {}", resourceName);
+            throw new IOException("Invalid JSON format");
```

Comment:
Why do you transform what was a ParseException to an IOException ?
Moreover, in the controller, there is already a catch on ParseException.